PoterekM/django-react-todo
# Set up api

## Description


This PR sets up the API:

- Rest framework has been added
- CORSheaders has been added
- localhost:3000 has been whitelisted 
- a serializer has been added to convert model instances to JSON so the frontend can work with the received data

The URLs file has been updated with a URL path for the API
- /todos/ - returns a list of all the Todo items. CREATE and READ operations can be performed here.
- /todos/id - returns a single Todo item using the id primary key. UPDATE and DELETE operations can be performed here.



## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| | :bug: Bug fix              |
| x | :sparkles: New feature     |
| | :hammer: Refactoring       |
|  | :100: Add tests            |
|   | :link: Update dependencies |
|  | :scroll: Docs              |


## After / New Behavior


Users can now complete CRUD functionality

Create and Read functionality: 
![create and read functionality](https://cdn.zappy.app/9f32010d36f55e9d1a63a547c4a1272a.gif)

Update and Delete functionality:
![](https://cdn.zappy.app/dce905a115e0945f0f17425472b6e264.gif)

## Testing Steps / QA Criteria

- Ensure your Python local environment is running
- `cd backend` and run `python manage.py runserver`

To test CREATE/READ functionality:

1. Navigate to http://localhost:8000/api/todos/
2. Enter a Title and Description for your todo list.

To test CREATE/READ functionality:
1. Navigate to http://localhost:8000/api/todos/todos/{id}
2. Test updating by modifying title or description
4. Delete item
